docs/guides/authentication.md

```diff
@@ -0,0 +1,76 @@
+# Authentication guide
+
+## Introduction
+
+The Medrunner API uses two methods of authentication: API keys and Discord OAuth2. API keys are used for server-to-server communication, while Discord OAuth2 is used for user authentication.
+
+## API keys
+
+To generate an API key, go to the [API Tokens section in your profile](https://portal.medrunner.space/profile) and click the "Create Token" button. You can provide a name for the key (Only used for your reference) and an optional expiration date. Once you click "Create", the key will be generated and displayed to you. You can then see all your keys in the table, indicating when they where last used and when they expire. You can delete a token if it is not in use anymore by clicking the trash icon.
+
+The API key is in reality a permanent refresh token, meaning you cannot use it directly to authenticate API calls. You first need to exchange it for an access token, follow the [Authenticate API calls](/guides/authentication#authenticate-api-calls) section for detailed steps.
+
+::: info
+Make sure to copy the generated key and store it in a safe place, as it will not be possible to see it again.
+:::
+
+## Discord OAuth2
+
+To authenticate users, the Medrunner API uses Discord OAuth2. This allows users to log in using their Discord account, and then you can get the Medrunner API tokens related to the discord account of the user.
+
+To get your first refresh tokens, you need to start by redirecting the user to the discord login page. You should then pass your discord client id and the redirect_url parameter (that should be the url where your application is). Discord sends back a code as a get request to the CALLBACK_URL provided.
+
+```ts
+window.location.replace(
+  `https://discord.com/oauth2/authorize?client_id=${DISCORD_CLIENT_ID}&scope=identify&response_type=code&redirect_uri=${CALLBACK_URL}/auth/register`,
+);
+```
+
+Once you have the discord code, you can exchange it for the refresh token by making a POST request to the Medrunner API at the `/auth/signin` endpoint with the discord code as the `code` parameter, and the CALLBACK_URL you used in the discord request as the `redirectUri` parameter.
+
+```ts
+await axios.get(`https://api.medrunner.space/auth/signin?code=${DISCORD_CODE}&redirectUri=${CALLBACK_URL}/auth`);
+```
+
+::: warning
+Make sure to use the exact same CALLBACK_URL you used in the discord request, otherwise the request will fail.
+:::
+
+To authenticate API calls, you now need to exchange it for an access token, follow the [Authenticate API calls](/guides/authentication#authenticate-api-calls) section for detailed steps.
```

Comment:
The user cannot do this, as the discord oath can only return to an explicitly whitelisted endpoint

Reply:
Removed discord OAuth2 mentions in the file in 1289ab2e23a0b185962533363eaa925ef6f50cb1